java/server/src/org/openqa/grid/web/servlet/api/v1/Proxy.java

```diff
@@ -39,7 +41,11 @@
     }
 
     final String proxyToFind = query.replaceAll("^/", "");
-    final RemoteProxy proxy = this.getRegistry().getAllProxies().getProxyById(proxyToFind);
+    RemoteProxy proxy = this.getRegistry().getAllProxies().getProxyById(proxyToFind);
```

Comment:
The problem is a proxy `id` can be anything a user wants it to be. Only by default or if the (user followed the default convention) will it be `http://{host}:{port}`

If you look at `java -jar selenium-server-standalone -role node -h` and the documentation for `id` it says this;

```shell
 -id
    <String> : optional unique identifier for the node. Defaults to the url
    of the remoteHost
```

Meaning it can be `http://ftp://$$$123/*hello/@123/rdp://` or any _crazy text_ I want it to be -- as long as it is unique. :)

So, we really need to one way hash the proxy id or similar and require the user to match all proxy `id`'s to their hash by making a GET on a `/proxies` resource (or I guess it could be the `/nodes` resource).. 

Come to think of it -- perhaps it is confusing having `/nodes` and `/proxy/{id}` .. we probably should combine these into one.

Something like;

`/proxies` -- get a list of all proxies
`/proxies/{id}` - get for the specified proxy

Similarly instead of `/session` and `/sessions` perhaps this would be better;

`/sessions` -- get a list of all sessions
`/sessions/{id}` - get for the specified session